### DISCUSSION TITLE
Capture Image as Jpeg

### DISCUSSION THREAD

**Discussion by garethem** [AxisCommunications/discussions#215] [ACAP] [answered]
I have an ACAP application which was written using the older ACAP 2 interfaces.
I need to port this to ACAP 3 or 4, but it uses the APIs from capture.h to capture a still image.

I have downloaded the acap-native-sdk docker image.

Can anyone provide a basic overview of still image capture using the new interfaces?

Thanks

**Comment by onie5** [comment, marked as answer]
The capture API was deprecated and replaced with libvdo. The method is almost similar to the capture methods.

Simplest way to capture a jpeg image:
1. Create a VdoMap object, supplying it with format (jpeg), width (horizontal resolution) and height (vertical resolution).
2. Take a snapshot with vdo_stream_snapshot() to a VdoBuffer object.
3.   You can then use VdoBuffer to get a VdoFrame which is the captured image.

The example vdostream is a good example but may need to experiment with it to better understand how it works.

Good luck.

**Comment by garethem** [comment]
Thanks. Yes, I had figured that out in the mean time. Works fine using the vdostream sample as a starting point
